Added a custom face for mouse highlighting

haji-ali opened the pull request.

clemera commented:
Thank you! Face names shouldn't end in "-face" as mentioned in the doc of `defface`. Can you also add a changelog? Before this gets merged we need to address #451 as otherwise you will get different faces for the extended highlight and the hovered candidate.

clemera commented:
With the above in mind the face also should inherit `selectrum-current-candidate` so these two are usually the same.

haji-ali commented:
How about this? I use the same face as `selectrum-current-candidate` except slanted.

clemera commented:
Sounded like a good idea but `:slant` is ignored by the `mouse-face`, see `(info "(elisp) Special Properties")`

clemera commented:
Maybe use `:underline` instead?

haji-ali commented:
Yeah, `:underline` could be a good replacement. I tried that and it works. 

However, I am seeing a weird issue in my setup. I am using marginlia which has the columns coloured differently. My `selectrum-current-candidate` simply changes the background so all the foreground colors remain unchanged even when a candidate is selected. But when I hover with the mouse, the foreground color changes depending on the foreground color of the character that the mouse was hovering before! See gif below.

Is this a known thing? Or am I doing something wrong?

<img src="https://user-images.githubusercontent.com/17233936/112017681-bdd78100-8b25-11eb-89ca-b00841c6565b.gif" align="left" width="400" >

clemera commented:
I have no idea why that is,  I haven't seen this before but I don't use the mouse regularly. I opened a separate issue for that.

haji-ali commented:
OK, I will investigate. In the meantime, I forced pushed a change with `:underline` instead of `:slant`.

clemera commented:
I'm okay to merge this before #451 gets addressed so if you add a changelog and remove the superfluous whitespace I can merge this ;)

haji-ali commented:
Ah sorry, I thought you meant the changelog in the commit message in your earlier message.
I removed the space and added an entry in the changelog. I wasn't sure if what I added was an "enhancement" or a "feature". I opted for the former.

minad commented:
Why is the face not inheriting from 'highlight? This would break fewer themes I guess?

clemera commented:
It inherits from `selectrum-current-candidate` so usually a theme hasn't to configure it in a special way unless it want to use a different face (which I assumed wouldn't be the usual case).

clemera commented:
I think using the same face as `selectrum-current-candidate` with an underline should work well for most? When using `highlight` and the theme doesn't use `highlight` for ``selectrum-current-candidate` there could be a an unexpected mismatch.

minad commented:
Okay, makes sense!

okamsn commented:
Somewhat related, I have asked Protesilaos Stavrou of the Modus themes to consider an option to make these faces more distinct.

https://gitlab.com/protesilaos/modus-themes/-/issues/203
